CONTRIBUTING.md

```diff
@@ -0,0 +1,223 @@
+
+For example:
+```ts
+const myFunction = async (myArg: MyArg): ThingResult => {
+    return doThing(myArg)
+}
+
+const myOtherFunction = async ({
+  myFirstArg,
+  mySecondArg,
+}: {
+  myFirstArg: MyFirstArg
+  mySecondArg: MySecondArg
+}): ThingResult => {
+  return doThing(myFirstArg, mySecondArg)
+}
+
+```
+
+In cases where the function has a complex set of arguments, the argument type can be defined in a declaration file and then assigned at the point of function implementation.
+
+```ts
+// in '.d.ts' file
+type MyFuncArgs = {
+    myFirstArg: MyFirstArg
+    mySecondArg: MySecondArg
+}
+
+// in other file
+const myFunction = async ({ myFirstArg, mySecondArg }: MyFuncArgs): ThingResult => {
+  return doThing(myFirstArg, mySecondArg)
+}
+```
+
+#### Defining objects with methods
+We use functional constructors to define certain types of objects that we can call method on. The intention here is to instantiate the object first and then call methods on that object with method-specific args to execute some functionality.
+
+For objects like these, the interface for the object is defined using a `type` declaration and methods are typed at this point.
+
+For example, our fee calculator is typed as follows:
+```ts
+// In '.d.ts' file
+type DepositFeeCalculator = {
+  onChainDepositFee({ amount, ratio }: onChainDepositFeeArgs): Satoshis
+  lnDepositFee(): Satoshis
+}
+
+// In implementation file
+export const DepositFeeCalculator = (): DepositFeeCalculator => {
+  const onChainDepositFee = ({ amount, ratio }: onChainDepositFeeArgs) => {
+    return toSats(Math.round(amount * ratio))
+  }
+
+  return {
+    onChainDepositFee,
+    lnDepositFee: () => toSats(0),
+  }
+}
+```
+
+Note that the top-level function arguments (for `DepositFeeCalculator`) would still be typed at the point of implementation like with any other function, and it is only the method signatures that are included in declaration files.
+
+#### Interfaces
+We use the `interface` keyword to define objects with methods that are intended to be implemented outside of the domain. This most often happens with service implementations like our `ILightningService` interface that gets implemented as the `LndService` function.
+
+Everything else about how we go about typing and implementing things with the `interface` keyword is the same as with how we handle "objects with methods" in our domain as described above.
+
+#### "Enums" via object constants
+Typescript's `enum` keyword has drawbacks that don't fully meet our typing needs. To get around this, we use the trick where we define our intended enum as a standard object and then import it into our type declarations.
+
+For example:
+```ts
+// In implementation file
+export const AccountStatus = {
+  Locked: "locked",
+  Active: "active",
+} as const
+
+// In declaration file
+type AccountStatus =
+  typeof import("./index").AccountStatus[keyof typeof import("./index").AccountStatus]
+```
+## Architecture
```

Comment:
we could use # for this title, and get one level above for all the below title (## instead of ### and so on)